### DISCUSSION TITLE
How to set KeepAlive for TCP connections

### DISCUSSION THREAD

**Discussion by PragmaTwice** [actix/actix-web#1835] [Q&A] [answered]
I found that `Tcp(usize)` in `enum KeepAlive` has been removed since about 2019.
Because I am using websocket, the option `Timeout(usize)` in `KeepAlive`  is useless.
How to use `SO_KEEPALIVE` for TCP connections in actix-web now?

**Comment by robjtede** [comment, marked as answer]
So you can delegate to the OS keepalives with https://docs.rs/actix-http/2.2.0/actix_http/enum.KeepAlive.html. Otherwise you'll need to use `socket2` to create a socket that has the exact duration set up and pass it to `HttpServer::listen`.
